lib/index.js

```diff
@@ -22,41 +24,24 @@ function setResponseHeader(request, header, value) {
   }
 }
 
-function setupPreResponseExt(server, options) {
-  server.ext('onPreResponse', (request, reply) => {
-    const requestLimit = request.plugins.patova && request.plugins.patova.limit;
-
-    if (requestLimit && requestLimit.conformant){
-      const headers = new RateLimitHeaders(
-          requestLimit.limit,
-          requestLimit.remaining,
-          requestLimit.reset);
-
-      Object.keys(headers).forEach(
-        key => setResponseHeader(request, key, headers[key]));
-    }
-    reply.continue();
-  });
-}
-
 function getMinimumLimit(limit1, limit2) {
   if (!limit1) { return limit2; }
   if (!limit2) { return limit1; }
 
-  if (limit1 && limit2.remaining > limit1.remaining) {
+  if (limit2.remaining > limit1.remaining) {
     return limit1;
   }
 
   return limit2;
 }
 
-function setupRateLimitEventExt(server, options) {
-  const event = options.event;
-  const extractKey = options.extractKey;
-  const onError = options.onError;
+function setupRateLimitEventExt(server, pluginOptions) {
+  const event = pluginOptions.event;
+  const onError = pluginOptions.onError;
 
-  const extractKeyAndTakeToken = function(limitd, request, reply, type) {
-    extractKey(request, reply, (err, key) =>{
+  const extractKeyAndTakeToken = function(options, request, reply, type) {
+    options.extractKey(request, reply, (err, key) =>{
+      const limitd = options.limitd;
       if (err) { return reply(err); }
 
       if (!limitd) {
```

Comment:
@dschenkelman do you want me to also remove this check ?
I'm not sure why it's there but I don't think `limitd` can ever be falsey at this stage